### PR TITLE
ngspice: fix the port for < 10.8

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Fixes `ngspice` after a breaking update, fixes what was broken in the earlier version.

@markemer @bpdegnan 

P. S. Likely fixups which are for < 10.7 should be used unconditionally, but let maintainers decide on that.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: science/ngspice/Portfile [hunk 1]
```diff
@@ -134,9 +161,16 @@ subport ngspice-lib {
...
+    if {${os.platform} eq "darwin" && ${os.major} <= 10} {
+        configure.cppflags-append \
+                                -D__NOEXTENSIONS__
+    }
+
     # Set CFLAGS and LDFLAGS
-    configure.cflags-append -m64 -O2 -Wall -I${prefix}/include/freetype2 -I${prefix}/include
-    configure.ldflags-append -m64 -L${prefix}/lib
+    configure.cflags-append     -O2 -Wall -I${prefix}/include/freetype2 -I${prefix}/include
+    configure.ldflags-append    -L${prefix}/lib
```
Comment (ryandesign) [inline science/ngspice/Portfile:173]: @bpdegnan Why is this still adding `-O2` or `-I${prefix}/include` or `-L${prefix}/lib`? I already explained in #25041 why we don't want that.

Comment (barracuda156) [reply science/ngspice/Portfile:173]: (I agree these should not be there, but I avoided to introduce changes beyond required ones.)

Comment (markemer) [reply science/ngspice/Portfile:173]: Let's pull it out - I missed that one too. Freetype has to stay thanks to some weird goings on with whatever the heck is going on with this build system.

File: science/ngspice/files/patch-fix-linking.diff [hunk 2]
```diff
@@ -0,0 +1,12 @@
+--- src/Makefile.in	2024-07-13 01:09:10.000000000 +0800
++++ src/Makefile.in	2024-08-31 13:25:22.000000000 +0800
+@@ -875,7 +875,8 @@
+ 	-e 's|@XSPICEINIT[@]|$(XSPICEINIT)|g' \
+ 	-e 's|@pkglibdir[@]|$(spinitpath)|g'
+ 
+-AM_CFLAGS = -static
++# MacOS linker does not need this.
++# AM_CFLAGS = -static
```
Comment (markemer) [inline science/ngspice/files/patch-fix-linking.diff:9]: I think this might be a weird old leftover, ngspice goes all the way back to SPICE3 essentially.

**Comment by markemer** [comment]
Aside from pulling out the other stuff we don't need, this looks good. Thanks @barracuda156. Your commitment to keeping new stuff running on old stuff running is pretty legendary.

File: science/ngspice/Portfile (hunk 1 above)
Comment (barracuda156) [reply science/ngspice/Portfile:173]: The build system inserts some silly flags here and there, maybe just patch those out.

**Comment by bpdegnan** [comment]
I have a Pismo with 10.4 that I only use to write papers.  I probably should check against that as a habit.  @barracuda156 thanks for being on top of it and @markemer, thanks for fixing it up.

**Pull request merged**